snarkos/ledger/server.rs

```diff
@@ -341,17 +351,43 @@ impl<N: Network> Server<N> {
         }
     }
 
-    /// Broadcasts the transaction to the ledger.
+    /// Send a program deployment transaction to the ledger
     async fn deploy_program(
         deployment: Deployment<N>,
         ledger: Arc<Ledger<N>>,
         ledger_sender: LedgerSender<N>,
     ) -> Result<impl Reply, Rejection> {
-        // Fetch the unspent records.
+        let additional_fee = Self::execute_additional_fee(ledger)?;
+        let transaction = Transaction::from_deployment(deployment.clone(), additional_fee).or_reject()?;
+        match ledger_sender.send(LedgerRequest::TransactionBroadcast(transaction)).await {
+            Ok(()) => Ok(reply::with_status(
+                reply::json(&json!({ "deployment": deployment })),
+                StatusCode::OK,
+            )),
+            Err(error) => Err(reject::custom(ServerError::Request(format!("{error}")))),
+        }
+    }
+
+    /// Send a program execution transaction to the ledger
+    async fn execute_program(
+        execution: Execution<N>,
+        ledger: Arc<Ledger<N>>,
+        ledger_sender: LedgerSender<N>,
+    ) -> Result<impl Reply, Rejection> {
+        let additional_fee = Self::execute_additional_fee(ledger)?;
+        let transaction = Transaction::from_execution(execution.clone(), Some(additional_fee)).or_reject()?;
+        match ledger_sender.send(LedgerRequest::TransactionBroadcast(transaction)).await {
+            Ok(()) => Ok(reply::with_status(reply::json(&json!({ "execution": execution })), StatusCode::OK)),
+            Err(error) => Err(reject::custom(ServerError::Request(format!("{error}")))),
+        }
+    }
+
+    /// Spends the record with the smallest value with at least `1 gate` as the fee
```

Comment:
"as the fee for the execution of a transaction." can be shortened to "as a transaction fee"